manifests/config/server/service.pp

```diff
@@ -37,19 +41,22 @@
   }
 
   if $service_ensure =~ /^(absent|false)$/ {
-    $changes = "rm Server/Service[#attribute/name='${name}']"
+    $changes = "rm Server/Service[#attribute/name='${service_name}']"
   } else {
-    if $class_name_ensure =~ /^(absent|false)$/ {
-      $_class_name = "rm Server/Service[#attribute/name='${name}']/#attribute/className"
-    } elsif $class_name {
-      $_class_name = "set Server/Service[#attribute/name='${name}']/#attribute/className ${class_name}"
+    if $class_name_ensure =~ /^(present|true)$/ {
+      if empty($class_name) {
+        fail('$class_name must be specified when $class_name_ensure is set to true or present')
+      }
+      $_class_name = "set Server/Service[#attribute/name='${service_name}']/#attribute/className ${class_name}"
+    } else {
+      $_class_name = "rm Server/Service[#attribute/name='${service_name}']/#attribute/className"
```

Comment:
Or actually, the original behavior isn't "do nothing," it is "make `class_name` absent" and the new behavior would create it. This could be incorrect for anyone depending on the previous behavior of making it absent, though it makes more sense to me...